workflow-examples/src/test/java/com/redhat/parodos/examples/integration/KubeapiWorkFlow.java

```diff
@@ -0,0 +1,130 @@
+package com.redhat.parodos.examples.integration;
+
+import com.redhat.parodos.sdk.api.ProjectApi;
+import com.redhat.parodos.sdk.api.WorkflowApi;
+import com.redhat.parodos.sdk.api.WorkflowDefinitionApi;
+import com.redhat.parodos.sdk.invoker.ApiClient;
+import com.redhat.parodos.sdk.invoker.ApiException;
+import com.redhat.parodos.sdk.invoker.Configuration;
+import com.redhat.parodos.sdk.model.*;
+import com.redhat.parodos.workflow.utils.CredUtils;
+import org.junit.Test;
+import org.springframework.http.HttpHeaders;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.junit.Assert.fail;
+
+public class KubeapiWorkFlow {
+
+	private final String projectName = "project-1";
+
+	private final String projectDescription = "Kubeapi example project";
+
+	private final String workflowName = "kubeapiWorkFlow";
+
+	private final String taskName = "kubeapiTask";
+
+	@Test
+	public void runFlow() throws ApiException {
+		ApiClient defaultClient = Configuration.getDefaultApiClient();
+
+		defaultClient.addDefaultHeader(HttpHeaders.AUTHORIZATION, "Basic " + CredUtils.getBase64Creds("test", "test"));
+
+		ProjectApi apiInstance = new ProjectApi(defaultClient);
+
+		try {
+			// RETRIEVE ALL PROJECTS AVAILABLE
+			List<ProjectResponseDTO> projects = apiInstance.getProjects();
+
+			// CHECK IF project ALREADY EXISTS
+			ProjectResponseDTO testProject = projects.stream()
+					.filter(prj -> projectName.equals(prj.getName()) && projectDescription.equals(prj.getDescription()))
+					.findAny().orElse(null);
+
+			// CREATE PROJECT IF NOT EXISTS
+			if (testProject == null) {
+				// DEFINE A PROJECT REQUEST
+				ProjectRequestDTO projectRequestDTO = ProjectRequestDTO.builder().name(projectName)
+						.description(projectDescription).build();
+
+				testProject = apiInstance.createProject(projectRequestDTO);
+			}
+
+			// ASSERT PROJECT IS PRESENT
+			projects = apiInstance.getProjects();
+			assertNotNull(projects.stream()
+					.filter(prj -> projectName.equals(prj.getName()) && projectDescription.equals(prj.getDescription()))
+					.findAny().orElse(null));
```

Comment:
this seems relevant only for the `if` block, else, if `testProject` isn't null, we repeat the same check  which isn't needed.

Reply:
The class [UseSDKExample](https://github.com/parodos-dev/parodos/blob/main/workflow-examples/src/test/java/com/redhat/parodos/examples/integration/UseSDKExample.java) it's just an **example** on how to start interact with SDK objects. It can't be considered as a valid implementation to copy/paste without applying improvements or corrections. The javadoc of that class says 
>  UseSDKExample is a dummy class to demonstrate very basic usage of @see workflow-service-sdk.

In other classes available in the same folder, you can see a different implementation ( [e.g. SimpleWorkFlow](https://github.com/parodos-dev/parodos/blob/14129d2ff83e2b8efc12ac03f15e36795459f2e4/workflow-examples/src/test/java/com/redhat/parodos/examples/integration/SimpleWorkFlow.java#L58) ) that asynchronously get all projects and creates a new one if needed.

Feel free to improve the code base if you think that there are uncovered scenarios/bugs/others.

Reply:
thanks @gciavarrini i will fix my code accordingly.

Reply:
fixed

Reply:
I'd like to suggest to use [ExamplesUtils.getProjectAsync](https://github.com/parodos-dev/parodos/blob/14129d2ff83e2b8efc12ac03f15e36795459f2e4/workflow-examples/src/test/java/com/redhat/parodos/examples/integration/SimpleWorkFlow.java#L58) : in this way there is no code duplication, it will use async requests e it will also wait for the service to be ready.

Please, use other utility methods as well (for example, when executing the workflow)

Reply:
can I get a one time waiver? I just want to merge it already.
BTW, it is not async. It is just waiting. Once the constraints for project name will be fixed this won't be needed. We will just create a project and if it already exists then it will fail

Reply:
Since @pkliczewski asked for the async request "feature", if it works for him then I'm ok.

BTW, the goal is twofold: it retries up to one minute to connect to `workflow-service` in order to be sure that the service has enough time to be up and running. It also use the async method provided by `ProjectApi` class [createProjectAsync](https://github.com/parodos-dev/parodos/blob/aee74dcd55658e59c5c02bff1512fcd9be4736c4/workflow-service-sdk/src/main/java/com/redhat/parodos/sdk/api/ProjectApi.java#L252)  which is different from [createProject](https://github.com/parodos-dev/parodos/blob/aee74dcd55658e59c5c02bff1512fcd9be4736c4/workflow-service-sdk/src/main/java/com/redhat/parodos/sdk/api/ProjectApi.java#L188)

Reply:
fixed